lib/editor/components/timetable/EditableCell.js

```diff
@@ -208,53 +207,11 @@ export default class EditableCell extends Component<Props, State> {
       const value = +data
       this._handleSave(value)
     } else {
-      // for time rendering
-      // FIXME replace with secondsAfterMidnightToHHMM?
       if (typeof data !== 'string') return this.cancel()
-      let time = data
-      let hours = 0
-      let parts
-      if (time.indexOf(':') > -1) {
-        // is a string with colons
-        parts = time.split(':')
-        hours = +parts[0]
-      } else {
-        // is a string without colons
-        if (time.length === 3) {
-          // format of `hmm`
-          parts = [time[0], time.substr(1, 3)]
-          hours = +time[0]
-        } else if (time.length === 4) {
-          // format of `HHmm`
-          parts = [time.substr(0, 2), time.substr(2, 4)]
-          hours = +time.substr(0, 2)
-        }
-      }
-
-      let greaterThan24 = false
-
-      // check for times greater than 24:00
-      // TODO: fix edge case of a stop time being 34 hours or more
-      // realistically this probably wouldn't happen, but a hypothetical
-      // case for this would be an airline flight that leaves at 23:59
-      // and flies for 13 hours before landing.  Another use-case is the
-      // trans-siberian railway that takes 7 days to travel from Moscow to
-      // Beijing.
-      if (parts && parts[0] && hours >= 24 && hours < 34) {
-        parts[0] = `0${hours - 24}`
-        greaterThan24 = true
-        time = parts.join(':')
-      }
-      const date = moment().startOf('day').format('YYYY-MM-DD')
-      const momentTime = moment(date + 'T' + time, TIMETABLE_FORMATS, true)
-      let value = momentTime.isValid() ? momentTime.diff(date, 'seconds') : null
-
-      if (greaterThan24 && momentTime.isValid()) {
-        value += 86400
-      }
-      // Check for valid time and new value
-      const timeIsValid = (time === '' || momentTime.isValid()) && value !== time
-      if (timeIsValid) this._handleSave(value)
+      const duration = moment.duration(data)
+      // $FlowFixMe: flow doesn't know about duration.isValid ensuring valueOf returns a number
+      const value = duration.isValid() ? duration.valueOf() / 1000 : null // ValueOf returns milliseconds
```

Comment:
```suggestion
      const value = duration.isValid() && duration.valueOf() / 1000 // ValueOf returns milliseconds
```
This can probably be shortened!

Reply:
You'll probably also have to change the condition to check for `false`